aws/rust-runtime/aws-config/src/profile/parser.rs

```diff
@@ -129,6 +134,35 @@ impl ProfileSet {
         self.profiles.keys().map(String::as_ref)
     }
 
+    /// Returns a string representation of the profiles, with syntax based on `kind`.
+    pub fn to_string(&self, kind: ProfileFileKind) -> String {
```

Comment:
I think we'll need a few caveats in this method:
1. It is going to delete any comments the user may have had
2. We need to test how it handles multi-line values, e.g. for S3:
```
[profile development]
s3 =
  max_concurrent_requests = 20
  max_queue_size = 10000
  multipart_threshold = 64MB
  multipart_chunksize = 16MB
  max_bandwidth = 50MB/s
  use_accelerate_endpoint = true
  addressing_style = path
```